lib/constructs/aws-serverless-s3-stack.ts

```diff
@@ -0,0 +1,115 @@
+     * @param {string} id - The scoped construct ID.
+     * @param {AwsServerlessS3StackProps} props - Properties for configuring the stack.
+     * @throws {Error} If any required properties are missing from props.
+     */
+    constructor(scope: Construct, id: string, props: AwsServerlessS3StackProps) {
+        super(scope, id, props);
+
+        // create master S3 bucket
+        const masterS3Bucket = this.createS3Bucket(this, `${props.resourcePrefix}-${props.masterBucketName}`, cdk.RemovalPolicy.DESTROY);
+
+        // apply life cycle rules for master bucket
+        this.createS3BucketLifecycleRule(masterS3Bucket, `${props.resourcePrefix}-${props.masterBucketName}-LifecycleRule`);
+
+        // assign the master bucket as an input for the next stack
+        this.masterS3BucketName = masterS3Bucket.bucketName;
+
+        // create pdf files S3 bucket
+        const pfdFilesBucket = this.createS3Bucket(this, `${props.resourcePrefix}-${props.pfdFilesBucketName}`, cdk.RemovalPolicy.DESTROY);
+
+        // apply life cycle rules for pdf files bucket
+        this.createS3BucketLifecycleRule(pfdFilesBucket, `${props.resourcePrefix}-${props.pfdFilesBucketName}-LifecycleRule`);
+
+        // assign the pdf files bucket as an input for the next stack
+        this.pfdFilesBucketName = pfdFilesBucket.bucketName;
+
+        // create json files S3 bucket
+        const jsonFilesBucket = this.createS3Bucket(this, `${props.resourcePrefix}-${props.jsonFilesBucketName}`, cdk.RemovalPolicy.DESTROY);
+
+        // apply life cycle rules for json files bucket
+        this.createS3BucketLifecycleRule(jsonFilesBucket, `${props.resourcePrefix}-${props.jsonFilesBucketName}-LifecycleRule`);
+
+        // assign the json files bucket as an input for the next stack
+        this.jsonFilesBucketName = jsonFilesBucket.bucketName;
+    }
+
+    /**
+     * Creates an S3 bucket with specified configurations.
+     *
+     * @param {Construct} scope - The scope in which to define this construct.
+     * @param {string} bucketName - The name of the S3 bucket to create.
+     * @param {cdk.RemovalPolicy} removalPolicy - The removal policy to apply to the bucket.
+     * @returns {s3.Bucket} The created S3 bucket.
+     * @throws {Error} If bucket creation fails due to invalid parameters or AWS API errors.
+     */
+    createS3Bucket(scope: Construct, bucketName: string, removalPolicy: cdk.RemovalPolicy): s3.Bucket {
+        return new s3.Bucket(scope, bucketName, {
+            bucketName: bucketName,
+            encryption: s3.BucketEncryption.S3_MANAGED,
+            blockPublicAccess: s3.BlockPublicAccess.BLOCK_ALL,
+            publicReadAccess: false,
+            removalPolicy: removalPolicy,
+            autoDeleteObjects: removalPolicy === cdk.RemovalPolicy.DESTROY,
+            accessControl: s3.BucketAccessControl.BUCKET_OWNER_FULL_CONTROL,
+            versioned: true, // Enable versioning
+            enforceSSL: true,
+        });
```

Comment:
**Suggestion:** Add error handling for the `createS3Bucket` method to catch and log any potential errors during bucket creation.
> 🏰  **High Priority** 
 > [Category: **`Error Handling`**, Importance: **`8`**]
> **Reason:** Adding error handling to the `createS3Bucket` method improves the reliability and debuggability of the code by catching and logging potential errors during bucket creation.

```suggestion
    createS3Bucket(scope: Construct, bucketName: string, removalPolicy: cdk.RemovalPolicy): s3.Bucket {
      try {
        return new s3.Bucket(scope, bucketName, {
          bucketName: bucketName,
          encryption: s3.BucketEncryption.S3_MANAGED,
          blockPublicAccess: s3.BlockPublicAccess.BLOCK_ALL,
          publicReadAccess: false,
          removalPolicy: removalPolicy,
          autoDeleteObjects: removalPolicy === cdk.RemovalPolicy.DESTROY,
          accessControl: s3.BucketAccessControl.BUCKET_OWNER_FULL_CONTROL,
          versioned: true, // Enable versioning
          enforceSSL: true,
        });
      } catch (error) {
        console.error(`Failed to create S3 bucket ${bucketName}:`, error);
        throw error;
      }
    }
```
- [ ]  **I have reviewed this code suggestion and addressed the relevant ones.**